src/fair_platform/utils/version.py

```
@@ -0,0 +1,119 @@
+        logger.debug(f"Failed to save update check timestamp: {e}")
+
+
+def get_latest_version_from_pypi() -> Optional[str]:
+    """Fetch the latest version from PyPI JSON API."""
+    try:
+        with httpx.Client(timeout=5.0) as client:
+            response = client.get("https://pypi.org/pypi/fair-platform/json")
+            response.raise_for_status()
+            data = response.json()
+            return data.get("info", {}).get("version")
+    except Exception as e:
+        logger.debug(f"Failed to fetch version from PyPI: {e}")
+        return None
+
+
+def check_for_updates() -> None:
+    """
+    Check for FAIR platform updates and notify the user if a new version is available.
+    
+    This function:
+    - Respects the FAIR_DISABLE_UPDATE_CHECK environment variable
+    - Caches checks for 24 hours
+    - Fails silently if offline or if PyPI is unavailable
+    - Only prints a message if a newer version is available
+    """
+    # Check if update checking is disabled
+    if os.getenv("FAIR_DISABLE_UPDATE_CHECK"):
+        return
+    
+    # Check if we should perform an update check
+    if not should_check_for_updates():
+        return
+    
+    # Get current and latest versions
+    current = get_current_version()
+    latest = get_latest_version_from_pypi()
+    
+    # Save the check timestamp (even if fetch failed)
+    if latest:
+        save_check_timestamp(latest)
+    
+    # Compare versions and notify if update available
+    if latest and latest != current:
+        # Simple version comparison (works for semantic versioning)
+        try:
+            from packaging import version as pkg_version
+            if pkg_version.parse(latest) > pkg_version.parse(current):
+                print(f"🔔 New version available: {latest} (current: {current})")
+                print("   Run: pip install -U fair-platform")
+        except Exception:
+            # If packaging is not available or comparison fails, just compare strings
```

This would fail for 0.10.0 and 0.2.0. Just fail silently